galsim/scene.py

```diff
@@ -568,6 +581,87 @@ def makeGalaxy(self, index=None, gal_type=None, chromatic=False, noise_pad_size=
         else:
             return gal_list[0]
 
+    def selectRandomIndex(self, n_random=1, rng=None, _n_rng_calls=False):
```

Comment:
This method reminds me of #825.  I think that issue is actually solved simply by something like

```python
result = np.zeros(n_random)
ud.generate(result)
return (result*n_choices).astype(int)
```

but it might be nice to also include the possibility of weights.  So I think it would nice to turn the logic here into a utility and mark #825 as completed too.